app.py

```diff
@@ -392,46 +392,64 @@ def _get_risk_level(debt_to_supply_ratio: float) -> str:
 
     streamlit.header("Detail of a loan")
     col1, col2, col3 = streamlit.columns(3)
-    with col1:
-        user = streamlit.text_input("User")
-        protocol = streamlit.text_input("Protocol")
-        users_and_protocols_with_debt = list(
-            loans_data.loc[
-                loans_data['Debt (USD)'] > 0,
-                ['User', 'Protocol'],
-            ].itertuples(index = False, name = None)
-        )
-        random_user, random_protocol = users_and_protocols_with_debt[numpy.random.randint(len(users_and_protocols_with_debt))]
-        if not user:
-            streamlit.write(f'Selected random user = {random_user}.')
-            user = random_user
-        if not protocol:
-            streamlit.write(f'Selected random protocol = {random_protocol}.')
-            protocol = random_protocol
-    loan = loans_data.loc[
-        (loans_data['User'] == user)
-        & (loans_data['Protocol'] == protocol),
-    ]
-    collateral_usd_amounts, debt_usd_amounts = src.main_chart.get_specific_loan_usd_amounts(loan = loan)
-    with col2:
-        figure = plotly.express.pie(
-            collateral_usd_amounts,
-            values='amount_usd',
-            names='token',
-            title='Collateral (USD)',
-            color_discrete_sequence=plotly.express.colors.sequential.Oranges_r,
-        )
-        streamlit.plotly_chart(figure, True)
-    with col3:
-        figure = plotly.express.pie(
-            debt_usd_amounts,
-            values='amount_usd',
-            names='token',
-            title='Debt (USD)',
-            color_discrete_sequence=plotly.express.colors.sequential.Greens_r,
-        )
-        streamlit.plotly_chart(figure, True)
-    streamlit.dataframe(loan)
+    try:
+        with col1:
+            user = streamlit.text_input("User")
+            protocol = streamlit.text_input("Protocol")
+            
+            users_and_protocols_with_debt = list(
+                loans_data.loc[
+                    loans_data['Debt (USD)'] > 0,
+                    ['User', 'Protocol'],
+                ].itertuples(index=False, name=None)
+            )
+        
+            random_user, random_protocol = users_and_protocols_with_debt[numpy.random.randint(len(users_and_protocols_with_debt))]
+            
+            if not user:
+                streamlit.write(f'Selected random user = {random_user}.')
+                user = random_user
+            if not protocol:
+                streamlit.write(f'Selected random protocol = {random_protocol}.')
+                protocol = random_protocol
+
+        loan = loans_data.loc[
+            (loans_data['User'] == user)
+            & (loans_data['Protocol'] == protocol),
+        ]
+
+        if loan.empty:
+            streamlit.warning(f"No loan found for user = {user} and protocol = {protocol}.")
+        else:
+            collateral_usd_amounts, debt_usd_amounts = src.main_chart.get_specific_loan_usd_amounts(loan=loan)
```

Comment:
You can wrap only this `src.main_chart.get_specific_loan_usd_amounts(loan=loan)` function in try/expect.